app/providers/operations/message/decryption_failed/index.js

```diff
@@ -0,0 +1,27 @@
+import DecryptionFailedMessagesPublicFields from "@sama/DTO/Response/message/decryption_failed/public_fields.js"
+
+class MessageDecryptionFailedOperation {
+  constructor(sessionService, userService, messageService, conversationService) {
+    this.sessionService = sessionService
+    this.userService = userService
+    this.messageService = messageService
+    this.conversationService = conversationService
+  }
+
+  async perform(ws, messageParams) {
+    const { cid, ids: mids } = messageParams
+
+    const currentUserId = this.sessionService.getSessionUserId(ws)
+    const currentUser = await this.userService.userRepo.findById(currentUserId)
+
+    await this.messageService.upsertMessageReadStatuses(cid, mids, currentUser.native_id, "decryption_failed")
```

Comment:
1) `upsertMessageReadStatuses -> upsertMessageStatus`
2) also, I do not see this method in message service `https://github.com/SAMA-Communications/sama-server/blob/4e8a26444c7fa366833efa993bf458a69111f295/app/providers/services/message/index.js`